autoscale/src/main/java/com/sequenceiq/periscope/service/MandatoryScalingAdjustmentService.java

```diff
@@ -0,0 +1,169 @@
+                    pollingUserCrn, stackV4Response, scalingAdjustmentParameters.getMandatoryStopStartDownscaleAdjustment());
+
+            scalingEventSender.sendScaleDownEvent(loadAlert, hostFqdnsToInstanceId.size(), hostsToDecommission, hostIdsWithServicesHealthy.size(), REGULAR);
+            LOGGER.info("Triggered mandatory adjustment ScaleDown for Cluster '{}', HostsToDecommission '{}', HostGroup '{}'",
+                    cluster.getStackCrn(), hostsToDecommission, policyHostGroup);
+        } else if (upscalable(scalingAdjustmentParameters.getMandatoryUpscaleAdjustment())) {
+            Integer targetScaleUpCount = scalingAdjustmentParameters.getMandatoryUpscaleAdjustment()
+                    .stream().map(adj -> IntStream.of(adj, loadAlertConfiguration.getMaxScaleUpStepSize()).min().getAsInt()).findFirst().orElse(0);
+
+            scalingEventSender.sendScaleUpEvent(loadAlert, existingClusterNodeCount, hostFqdnsToInstanceId.size(), hostIdsWithServicesHealthy.size(),
+                    targetScaleUpCount);
+            LOGGER.info("Triggered mandatory adjustment ScaleUp for Cluster '{}', NodeCount '{}', HostGroup '{}'", cluster.getStackCrn(),
+                    targetScaleUpCount, policyHostGroup);
+        } else if (downscalable(scalingAdjustmentParameters.getMandatoryDownscaleAdjustment())) {
+            List<String> hostsToDecommission = collectDownscaleRecommendations(hostFqdnsToInstanceId, emptyList(),
+                    pollingUserCrn, stackV4Response, scalingAdjustmentParameters.getMandatoryDownscaleAdjustment());
+
+            scalingEventSender.sendScaleDownEvent(loadAlert, hostFqdnsToInstanceId.size(), hostsToDecommission, hostIdsWithServicesHealthy.size(), REGULAR);
+            LOGGER.info("Triggered mandatory adjustment ScaleDown for Cluster '{}', HostsToDecommission '{}', HostGroup '{}'",
+                    cluster.getStackCrn(), hostsToDecommission, policyHostGroup);
+        }
+    }
+
+    private boolean upscalable(Optional<Integer> adjustmentParam) {
+        return adjustmentParam.isPresent() && isCoolDownTimeElapsed(cluster.getStackCrn(), "scaled-up",
+                loadAlertConfiguration.getScaleUpCoolDownMillis(), cluster.getLastScalingActivity());
+    }
+
+    private boolean downscalable(Optional<Integer> adjustmentParam) {
+        return adjustmentParam.isPresent() && isCoolDownTimeElapsed(cluster.getStackCrn(), "scaled-down",
+                loadAlertConfiguration.getScaleDownCoolDownMillis(), cluster.getLastScalingActivity());
+    }
+
+    private List<String> collectDownscaleRecommendations(Map<String, String> hostFqdnsToInstanceId, List<String> stoppedHostIds,
+            String pollingUserCrn, StackV4Response stackResponse, Optional<Integer> mandatoryDownscaleCount) {
+        try {
+            YarnScalingServiceV1Response yarnResponse = yarnMetricsClient.getYarnMetricsForCluster(cluster, stackResponse, policyHostGroup,
```

Comment:
Before invoking YARN - check if the existing stopped instances are adequate. There's no need to invoke YARN in this case.

Reply:
Updated this